SocialCareCaseViewerApi.Tests/V1/Gateways/Database/GetAllResidentsTests.cs

```diff
@@ -0,0 +1,506 @@
+
+            var listOfPersons = _classUnderTest.GetResidentsBySearchCriteria(cursor: 0, limit: 20, postcode: "E8 1DY").ToList();
+
+            var personUnderTest = listOfPersons
+                .First(p => p.MosaicId.Equals(person.Id.ToString()));
+
+            personUnderTest.PhoneNumber.Should().ContainEquivalentOf(phoneNumber.ToResponse());
+        }
+
+        [Test]
+        public void WithBirthdayQueryParameterReturnsMatchingResidents()
+        {
+            var person1 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            var person2 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            var person3 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+
+            person3.DateOfBirth = person1.DateOfBirth;
+
+            DatabaseContext.Persons.AddRange(new List<Person> { person1, person2, person3 });
+            DatabaseContext.SaveChanges();
+
+            var listOfPersons = _classUnderTest.GetResidentsBySearchCriteria(cursor: 0, limit: 20, dateOfBirth: person1.DateOfBirth.Value.ToString("yyyy-MM-dd")).ToList(); //Format passed be FE
+
+            listOfPersons.Count.Should().Be(2);
+            listOfPersons.Should().ContainEquivalentOf(DatabaseContext.Persons.First(x => x.Id == person1.Id).ToResidentInformationResponse());
+            listOfPersons.Should().ContainEquivalentOf(DatabaseContext.Persons.First(x => x.Id == person3.Id).ToResidentInformationResponse());
+        }
+
+        [Test]
+        public void WithContextFlagQueryParameterReturnsMatchingResidents()
+        {
+            var person1 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            var person2 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            var person3 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+
+            person3.AgeContext = person1.AgeContext;
+
+            DatabaseContext.Persons.AddRange(new List<Person> { person1, person2, person3 });
+            DatabaseContext.SaveChanges();
+
+            var listOfPersons = _classUnderTest.GetResidentsBySearchCriteria(cursor: 0, limit: 20, contextflag: person1.AgeContext).ToList();
+
+            listOfPersons.Count.Should().Be(2);
+            listOfPersons.Should().ContainEquivalentOf(DatabaseContext.Persons.First(x => x.Id == person1.Id).ToResidentInformationResponse());
+            listOfPersons.Should().ContainEquivalentOf(DatabaseContext.Persons.First(x => x.Id == person3.Id).ToResidentInformationResponse());
+        }
+
+        [Test]
+        public void WithNameAndPostCodeQueryParameterReturnsMatchingResident()
+        {
+            var person1 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            person1.FirstName = "ciasom";
+
+            var person2 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            person2.FirstName = "wrong name";
+
+            var person3 = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            person3.FirstName = "ciasom";
```

Comment:
Looks like `CreatePersonDatabaseEntity` takes in first name as a parameter, could this be refactored to use it?

Reply:
yes, forgot to tidy that up. I've updated it now